crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -861,7 +957,11 @@ where
         &self,
         range: impl RangeBounds<TxNumber>,
     ) -> ProviderResult<Vec<Receipt>> {
-        self.database.receipts_by_tx_range(range)
+        self.entries_by_tx_range(
+            range,
+            |range| self.database.receipts_by_tx_range(range),
+            |block, index| Ok(block.executed_block_receipts()[index].clone()),
```

Comment:
how do we know that the result of `block.executed_block_receipts` will have at least `index + 1` elements? otherwise `block.executed_block_receipts()[index]` would panic?

same in the other closures to query in-memory above

Reply:
because number of receipts is equal to the number of transactions, and the number of transactions is at least `index + 1` because we pass the block and index from the loop iterating over transactions https://github.com/paradigmxyz/reth/blob/65f26a0a8ddb97e78b3ae0caf9b243ba0911df8c/crates/storage/provider/src/providers/blockchain_provider.rs#L263